draft-proxied-svcb-headers.md

```diff
@@ -17,132 +17,545 @@ author:
+through proxies supporting this specification while reducing the need
+for performing additional DNS lookups:
+
+1) Clients provide additional information in HTTP request header fields,
+   allowing the proxy to perform appropriate SVCB lookups alongside
+   its AAAA and A lookups.
+2) Proxies return HTTP response header fields specified in this
+   document while opportunistically establishing connections.
+3) Services with no SVCB records will be indicated as such
+   by the proxy, allowing clients can proceed with the opportunistically
+   established connection.  For services with SVCB records,
+   the proxy will provide enough information to allow clients
+   to decide whether they can proceed with using the connection
+   or whether the client needs to establish a new connection
+   through the proxy to the alternative endpoint specified
+   in a SVCB record.  
+4) For subsequent CONNECT* requests, clients provide information
+   about the service name.  The proxy uses this to provide
+   refreshed SVCB records, and the proxy also continues to
+   provide additional information to the client about how
+   it established any given connection.
+
+*TODO*: bikeshed on the names for these headers.
+
+This document defines a new request header:
+
+* Proxy-DNS-Request: requests that a proxy provide information
+  from DNS lookups in its response, including AAAA/A/CNAME
+  information used by the proxy in making its connection,
+  as well as SVCB records for a name specified in this header.
+
+This document also defines two new response headers:
+
+* Proxy-DNS-SVCB: encodes SVCB records associated with requested name.
+* Proxy-DNS-Used: encodes DNS information used by the Proxy for making this connection.
+
+*TODO*: do we need a Terminology section, or a reference to Section 1.4 of {{SVCB}} as well as {{!RFC8499}}?
 
 ## Requirements
 
 {::boilerplate bcp14}
 
-# SVCB Request Header Field {#svcb-params-request}
 
-Clients can request SVCB parameters with the Structured Header
-{{!RFC8941}} "DNS-SVCB-Keys". Its value MUST
-be an sf-list whose members are sf-integer items that MUST NOT contain parameters. Its
-ABNF is:
+# Definitions of request and response header fields
+
+## Proxy-DNS-Request Request Header Field {#proxy-dns-request}
+
+Clients can send the Structured Header {{!RFC8941}} "Proxy-DNS-Request"
+to request that proxies send them DNS-related information.
+Its value MUST be an sf-item whose type is sf-string followed by a set of parameters.
+
+Its ABNF is:
 
 ~~~ abnf
-DNS-SVCB-Keys = sf-list
+Proxy-DNS-Request = sf-item
 ~~~
 
-Each list member corresponds to the numeric version of an SvcParamKey.
+The item value is the name of a SVCB DNS query name.  This is the service
+name, possibly prepended with a label ({{SVCB}} Section 2.3).
+For example, "example.com" or "_foo.example.com" or "_8443._bar.api.example.com".
+
+*TODO*: Should this have a trailing "." to be an FQDN?
```

Comment:
I'm a bit ambivalent here. I assume that the proxy can always append the "." when doing the lookup.

Reply:
I'm also ambivalent here.  We should pick one and specify, though.  Or specify that if the trailing "." is missing that the consumer should assume its presence and add it.  (The same applies for all of these containing DNS names.)